lib/node_modules/@stdlib/plot/table/unicode/benchmark/benchmark.render.js

```diff
@@ -0,0 +1,142 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var Float64Array = require( '@stdlib/array/float64' );
+var array = require( '@stdlib/ndarray/array' );
+var pkg = require( './../package.json' ).name;
+var UnicodeTable = require( './../lib' );
+
+
+// FUNCTIONS //
+
+/**
+* Creates a benchmark function.
+*
+* @private
+* @param {PositiveInteger} rows - number of rows
+* @param {PositiveInteger} columns - number of columns
+* @param {UnicodeSparkline} table - table instance
+* @returns {Function} benchmark function
+*/
+function createBenchmark( rows, columns, table ) {
+	return benchmark;
+
+	/**
+	* Generates table data.
+	*
+	* @private
+	* @returns {ndarray} table data
+	*/
+	function data() {
+		var data;
+		var i;
+
+		data = new Float64Array( rows * columns );
+		for ( i = 0; i < data.length; i++ ) {
+			data[ i ] = randu();
+		}
+		data = array( data, {
+			'shape': [ rows, columns ]
+		});
+		return data;
+	}
+
+	/**
+	* Generates table headers.
+	*
+	* @private
+	* @returns {Float64Array} table data
+	*/
+	function headers() {
+		var data;
+		var i;
+
+		data = new Float64Array( columns );
+		for ( i = 0; i < data.length; i++ ) {
+			data[ i ] = randu();
+		}
+		return data;
+	}
+
+	/**
+	* Benchmark function.
+	*
+	* @private
+	* @param {Benchmark} b - benchmark instance
+	*/
+	function benchmark( b ) {
+		var str;
+		var i;
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			str = table.setData( data(), headers() ).render();
```

Comment:
You definitely do not want to be generating fresh data and headers for every benchmark iteration. Otherwise, you confound results.